android-components/components/feature/search/src/main/java/mozilla/components/feature/search/telemetry/BaseSearchTelemetry.kt

```diff
@@ -21,20 +25,47 @@ import mozilla.components.support.base.facts.collect
 import mozilla.components.support.base.log.logger.Logger
 import mozilla.components.support.ktx.kotlinx.coroutines.flow.filterChanged
 import org.json.JSONObject
+import java.io.File
+
+internal const val COLLECTION_NAME = "search-telemetry-v2"
 
 /**
  * Main configuration and functionality for tracking ads / web searches with specific providers.
  */
-abstract class BaseSearchTelemetry {
+abstract class BaseSearchTelemetry(private val file: File) {
+    @VisibleForTesting
+    internal var providerList: List<SearchProviderModel>? = emptyList()
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PROTECTED)
+    internal fun initializeProviderList(): Job {
+        val fetcher = SerpTelemetryFetcher(file, COLLECTION_NAME)
+
+        return CoroutineScope(Dispatchers.IO).launch {
+            providerList = fetcher.fetchSearchProviders()
+        }
+    }
 
     /**
-     * Install the web extensions that this functionality is based on and start listening for updates.
+     * Finds provider among list of providers that matches regex in url.
      */
-    abstract fun install(engine: Engine, store: BrowserStore)
+    @VisibleForTesting(otherwise = VisibleForTesting.PROTECTED)
+    fun getProviderForUrl(url: String): SearchProviderModel? {
```

Comment:
drive-by: Did `getProviderForUrl` fun need to change? E.g. the added brackets and return statement as opposed to inline

Reply:
It didn't, guess it changed a few times and ended up here. I can revert it. Thanks for catching this!